### PR TITLE
feat: load/save sdk settings support

### PR DESCRIPTION
**Pull request by mrehan27**
### Changes

- Support to bridge native code for AmiApp
- Support for saving/loading Customer.io SDK configurations
- Support for setting keys privately
- Support for saving profile identifier locally
- ViewLogs and Settings screen updates

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,6 @@ app.*.map.json
 /android/app/debug
 /android/app/profile
 /android/app/release
+
+# File to keep Customer.IO keys private
+/assets/secrets/keys.json
```
Comment (Shahroz16) [inline .gitignore:50]: Can't we have an `env` file?

File: android/app/src/main/kotlin/io/customer/amiapp_flutter/MainActivity.kt [hunk 2]
```diff
@@ -1,6 +1,78 @@
...
+    private fun clearLogs(): Any? {
+        amiAppLogger.clearLogs()
+        return null
+    }
+
+    private fun getUserAgent(): String? {
+        return "User agent will be shown here"
+    }
+
+    private fun updateDeviceToken(): Any? {
```
Comment (Shahroz16) [inline android/app/src/main/kotlin/io/customer/amiapp_flutter/MainActivity.kt:59]: hmm not sure what is this for?

File: lib/src/customer_io.dart [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+class CustomerIOConfigurations {
+  String siteId;
+  String apiKey;
+  String? organizationId;
+  Region? region;
+  String? trackingUrl;
+  String? gistEnvironment;
+  int? backgroundQueueSecondsDelay;
+  int? backgroundQueueMinNumberOfTasks;
+  bool? featureEnablePush;
```
Comment (Shahroz16) [inline lib/src/customer_io.dart:235]: what does this config do?

File: .gitignore (hunk 1 above)
Comment (mrehan27) [reply .gitignore:50]: Yeah, we can rename it to `env`, will update it.

File: android/app/src/main/kotlin/io/customer/amiapp_flutter/MainActivity.kt (hunk 2 above)
Comment (mrehan27) [reply android/app/src/main/kotlin/io/customer/amiapp_flutter/MainActivity.kt:59]: This is temporary method to update token as it is async, will be removing it and updating `getToken` to future in upcoming PRs

File: lib/src/customer_io.dart (hunk 3 above)
Comment (mrehan27) [reply lib/src/customer_io.dart:235]: I copied it from designs initially, but I don't think we need this, will remove it in next PR.

File: .gitignore (hunk 1 above)
Comment (mrehan27) [reply .gitignore:50]: I just saw [flutter_dotenv](https://pub.dev/packages/flutter_dotenv) that you used, will replace it in next PR.

File: android/app/src/main/kotlin/io/customer/amiapp_flutter/MainActivity.kt (hunk 2 above)
Comment (mrehan27) [reply android/app/src/main/kotlin/io/customer/amiapp_flutter/MainActivity.kt:59]: As discussed, we'll replace it once changes from native SDK are live.

**Review by Shahroz16 — APPROVED** [review]

**Pull request merged**